Mathlib/Data/Finset/Card.lean

```diff
@@ -579,6 +579,21 @@ lemma exists_subsuperset_card_eq (hst : s ⊆ t) (hsn : #s ≤ n) (hnt : n ≤ #
 lemma exists_subset_card_eq (hns : n ≤ #s) : ∃ t ⊆ s, #t = n := by
   simpa using exists_subsuperset_card_eq s.empty_subset (by simp) hns
 
+lemma of_cardinality_between_of_disjoint (hs : #s < n) (hn : n ≤ #s + #t) (hst : Disjoint s t) :
+    ∃ (x : Finset α) (hxs : Disjoint x s), #(x.disjUnion s hxs) = n ∧ Nonempty x := by
```

Comment:
I think we should be satisfied with this statement:
```suggestion
lemma of_cardinality_between_of_disjoint (hs : #s ≤ n) (hn : n ≤ #s + #t) (hst : Disjoint s t) :
    ∃ (x : Finset α) (hxs : Disjoint x s), #(x.disjUnion s hxs) = n := by
```
because can be easily deduced from this:
```
import Mathlib
open Finset in
example {n} {α} (s x : Finset α) (hxs : Disjoint x s) (eq : #(x.disjUnion s hxs) = n)
    (lt : #s < n) : x ≠ ∅ := by
  rintro rfl; rw [← eq] at lt; simp at lt
```
(Note I've replaced `Nonempty x` with `x ≠ ∅`.)

Reply:
For the purposes of #19607 I prefer the stronger version.

Reply:
Do you want to have both lemmas?

Reply:
Yes please; and please use the `x ≠ ∅` statement: from the lemmas around [Finset.nonempty_iff_ne_empty](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Finset/Empty.html#Finset.nonempty_iff_ne_empty) I think statements involving ∅ are considered the normal form.

Reply:
Pinging @YaelDillies for a second opinion on whether that or `Nonempty` is preferred.